Q-8. The anxiety levels of 10 participants were measured before and after a new therapy.
The scores are not normally distributed. Use the Wilcoxon signed-rank test to test whether
the therapy had a significant effect on anxiety levels. The data is given below: Participant
Before therapy After therapy Difference

In [7]:
import pandas as pd
import seaborn as sns
from scipy.stats import wilcoxon

In [9]:
data = {
    'Participant': [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
    'Before therapy': [10, 8, 12, 15, 6, 9, 11, 7, 14, 10],
    'After therapy': [7, 6, 10, 12, 5, 8, 9, 6, 12, 8],
    'Difference': [-3, -2, -2, -3, -1, -1, -2, -1, -2, -2]
}

In [10]:
df = pd.DataFrame(data)
df

Participant  Before therapy  After therapy  Difference
0            1              10              7          -3
1            2               8              6          -2
2            3              12             10          -2
3            4              15             12          -3
4            5               6              5          -1
5            6               9              8          -1
6            7              11              9          -2
7            8               7              6          -1
8            9              14             12          -2
9           10              10              8          -2

In [12]:
# Perform the Wilcoxon signed-rank test

statistic, p_value = wilcoxon(df['Before therapy'], df['After therapy'])

# Print the test statistic and p-value
print("Test Statistic:", statistic)
print("P-value:", p_value)

Test Statistic: 0.0
P-value: 0.001953125


The test statistic for the Wilcoxon signed-rank test is 0.0, and the corresponding p-value is 0.001953125.

Since the p-value is less than the significance level of 0.05, we can reject the null hypothesis that the therapy has no significant effect on anxiety levels. The results suggest that the therapy has a significant effect in reducing anxiety levels.